app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -128,22 +135,61 @@ private void loadSavedInstanceState(Bundle savedInstanceState) {
     }
 
     private void setupUpdateTask(boolean isRotation) {
-        updateTask = UpdateTask.getRunningInstance();
-
-        if (updateTask != null) {
-            try {
-                updateTask.registerTaskListener(this);
-            } catch (TaskListenerRegistrationException e) {
-                Log.e(TAG, "Attempting to register a TaskListener to an already " +
-                        "registered task.");
-                uiController.errorUiState();
+        if (isAutoUpdateInProgress()) {
+            connectToUpdateWorker();
+        } else {
+            updateTask = UpdateTask.getRunningInstance();
```

Comment:
Right with the difference being `UpdateTask` is the reliable mechanism for foreground updates while `UpdateWorker` is only used for background ones. WorkManager is the replacement for traditional Android services kind of work and is not recommended for doing one off forground task that needs to happen on a user trigger as there is no way we can schedule a WorkManager task immediately.

Reply:
Got it, that's frustrating in terms of the framework (especially since they have different reporting mechanisms back to the UI layer). The design makes sense given those constraints